src/Model/CachingIterator.php

```diff
@@ -87,7 +90,9 @@ public function count()
     #[ReturnTypeWillChange]
     public function current()
     {
-        return current($this->items);
+        /* Fetching current item */
+        $currentItem = current($this->items);
+        return $currentItem !== false ? $currentItem[self::FIELD_VALUE] : false;
```

Comment:
Coding standards will require a blank line before `return` statements. See: https://github.com/mongodb/mongo-php-library/runs/5471596028?check_suite_focus=true#step:9:1

This didn't show up earlier as I had to approve the workflow run for your first pull request; however, you might have caught this locally by running `vendor/bin/phpcs` (and `vendor/bin/phpcbf` could have fixed it for you).